### PR TITLE
libckteec: session functions (open/close/get info)

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: libckteec/src/pkcs11_token.c [hunk 1]
```diff
@@ -392,3 +392,165 @@ CK_RV ck_token_mechanism_info(CK_SLOT_ID slot, CK_MECHANISM_TYPE type,
...
+				    ctrl, out, &out_size);
+
+	if (rv != CKR_OK || out_size != out->size) {
+		if (rv == CKR_OK)
+			rv = CKR_DEVICE_ERROR;
+		goto out;
+	}
+
+	ta_info = (struct pkcs11_session_info *)out->buffer;
+        info->slotID = ta_info->slot_id;
```
Comment (jenswi-linaro) [inline libckteec/src/pkcs11_token.c]: Strange indentation

Comment (etienne-lms) [reply libckteec/src/pkcs11_token.c]: fixed. thanks.

**Comment by etienne-lms** [comment]
Indentation fixed. Typos in pkcs11_ta.h inline comments fixed.
A-b tag applied.

**Comment by jforissier** [comment]
There's a line >80 characters that could be fixed.

**Comment by etienne-lms** [comment]
> There's a line >80 characters that could be fixed.

Addressed, thanks.
Remains the few checkpatch reports:
```
d8362de libckteec: sync with pkcs11 ta api for session commands
CHECK: Prefer using the BIT macro
#87: FILE: libckteec/include/pkcs11_ta.h:468:
+#define PKCS11_CKFSS_RW_SESSION			(1 << 1)

CHECK: Prefer using the BIT macro
#88: FILE: libckteec/include/pkcs11_ta.h:469:
+#define PKCS11_CKFSS_SERIAL_SESSION		(1 << 2)

83d75ee libckteec: debug string helpers for session flags and state
WARNING: Possible switch case/default not preceded by break or fallthrough comment
#60: FILE: libckteec/src/ck_debug.c:300:
+	default:

total: 0 errors, 1 warnings, 0 checks, 39 lines checked
```

**Comment by etienne-lms** [comment]
updated with P-R reference fixed in 1st commit message.
tag still applied (and applicable:)

**Pull request merged**